cpp/include/cudf/copying.hpp

```diff
@@ -85,6 +97,26 @@ table empty_like(table const& t);
  */
 table allocate_like(table const& t, bool allocate_mask_if_exists = true, cudaStream_t stream = 0);
 
+/**
+ * @brief Creates a table of columns with the specified size and same type as
+ * the input.
+ *
+ * Creates the `gdf_column` objects, and allocates underlying device memory for
+ * each column matching the input columns
+ *
+ * @note It is the caller's responsibility to free each column's device memory
+ * allocation in addition to deleting the `gdf_column` object for every column
+ * in the new table.
+ *
+ * @param t The table to emulate
+ * @param size The size of the columns to allocate
+ * @param allocate_mask_if_exists Optional whether or not to allocate the bitmask for each column if it exists in the corresponding input column
+ * @param stream Optional stream in which to perform allocations
+ * @return table A table of columns with same type and allocation size as input
```

Comment:
A table of columns with same type and specified size

Reply:
Done, thanks.